### ISSUE TITLE
Swipe back is propagated on the screen behind on Android

### ISSUE THREAD

**Issue by durdevic** [react-navigation/react-navigation#8745]
**Current Behavior**
I have gestureEnabled on Stack.Navigator set to **true** and I'm disabling it on the WebView screen like this.

```
function WebViewScreen({ route, navigation }) {

    useEffect(() => {
        navigation.setOptions({ gestureEnabled: false });
    }, []);
...
```

**Expected Behavior**
The swipe back is propagated on Android, but not on iOS. It should not be propagated on Android.

**How to reproduce**

Test on Android and iOS. 
Instructions: 
_Try swiping on the screen back from here (a couple of times preferably), then click on just go back button. You should see "This is the first screen", on Android and "This is the second screen" on iOS. I put them also on the last page of the snack_

https://snack.expo.io/@ukiand1/swipe-propagation

Videos from the Snack:

![Screen Recording 2020-08-19 at 12 15 05](https://user-images.githubusercontent.com/2753836/90627554-67673480-e21c-11ea-96a2-6fe9a4293fbe.gif)
![Screen Recording 2020-08-19 at 12 13 52](https://user-images.githubusercontent.com/2753836/90627572-6d5d1580-e21c-11ea-8dd3-4eaaaf32fad4.gif)

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | Android
| @react-navigation/native       | ^5.1.7
| @react-navigation/stack        | ^5.9.0
| react-native-gesture-handler   | ^1.6.1
| react-native-safe-area-context |^0.7.3
| react-native-screens           | ^2.4.0
| react-native                   | ^0.62.2
| expo                           |
| node                           | 
| npm or yarn                    | 6.12.0

**Comment by durdevic** [comment]
This is the configuration in Snack with the same behaviour 

```
{
  "dependencies": {
    "react-native-paper": "3.6.0",
    "react-native-screens": "^2.0.0",
    "@react-navigation/stack": "^5.9.0",
    "@react-navigation/native": "^5.7.3",
    "react-native-safe-area-context": "^0.6.0",
    "@react-native-community/masked-view": "^0.1.7"
  }
}
```

**Comment by durdevic** [comment]
Solved in react-native-gesture-handler: 1.8.0
